File: src/main/java/com/iot/temperature/TemperatureExpiryPolicy.java
package com.iot.temperature;

import javax.cache.expiry.Duration;
import javax.cache.expiry.ExpiryPolicy;

public class TemperatureExpiryPolicy implements ExpiryPolicy{

	@Override
	public Duration getExpiryForCreation() {
		return Duration.TEN_MINUTES;
	}

	@Override
	public Duration getExpiryForAccess() {
		return Duration.TEN_MINUTES;
	}

	@Override
	public Duration getExpiryForUpdate() {
		return Duration.TEN_MINUTES;
	}

}

File: src/main/kotlin/temperature/TemperatureExpiryPolicy.kt
package temperature

import javax.cache.expiry.Duration
import javax.cache.expiry.ExpiryPolicy

class TemperatureExpiryPolicy : ExpiryPolicy {
    override fun getExpiryForCreation(): Duration {
        return Duration.TEN_MINUTES
    }

    override fun getExpiryForAccess(): Duration {
        return Duration.TEN_MINUTES
    }

    override fun getExpiryForUpdate(): Duration {
        return Duration.TEN_MINUTES
    }
}